lib/Sema/CSGen.cpp

```diff
@@ -3769,31 +3769,14 @@ bool swift::areGenericRequirementsSatisfied(
   return CS.solveSingle().hasValue();
 }
 
-static bool canSatisfy(Type type1, Type type2, bool openArchetypes,
+bool swift::canSatisfy(Type type1, Type type2, bool openArchetypes,
```

Comment:
Can this be moved to lib/IDE as well?

Reply:
It's quite difficult. The implementation of `canSatisfy()` invokes `TypeChecker::typesSatisfyConstraint` which is not part of sema APIs we can call from libIDE.

Reply:
Sorry, I meant move it to IDETypeChecking.cpp